components/autocomplete/index.js

```diff
@@ -162,14 +163,18 @@ export class Autocomplete extends Component {
 	}
 
 	select( option ) {
+		const { getBlockType } = this.props;
 		const { open, range, query } = this.state;
-		const { onSelect } = open || {};
+		const { getOptionResult } = open || {};
 
 		this.reset();
 
-		if ( onSelect ) {
-			const replacement = onSelect( option.value, range, query );
-			if ( replacement !== undefined ) {
+		if ( getOptionResult ) {
+			const replacement = getOptionResult( option.value, range, query );
+
+			if ( replacement && typeof replacement === 'object' && getBlockType( replacement.name ) ) {
```

Comment:
Is there maybe an existing API method which validated if the input is a block? There is a similar logic in here:
https://github.com/WordPress/gutenberg/blob/master/blocks/block-edit/index.js#L22

We might want to extract method and use it in both places.

Reply:
Agreed. I'll look at that.

Reply:
I took a look and am wondering whether it is worth it to a function for the check. The only thing in common is that both make sure a truthy value is returned from `getBlockType`. Defining an `isBlockType` function in `blocks/api/registration` may allow us to be more direct though.

Currently, I simply moved the conditions from `components/Autocomplete` to `blocks/contextual-autocomplete` in the form of the `isReplacement` prop.